File: gymgenie-mobile/composeApp/src/androidMain/kotlin/com/asc/gymgenie/feature/workouts/components/MuscleGroupFilterChips.kt
package com.asc.gymgenie.feature.workouts.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.asc.gymgenie.ui.theme.AccentOrange
import com.asc.gymgenie.ui.theme.DeepInk
import com.asc.gymgenie.ui.theme.SoftCard

private val MuscleGroups: List<Pair<String?, String>> = listOf(
    null to "Все",
    "CHEST" to "Грудь",
    "BACK" to "Спина",
    "SHOULDERS" to "Плечи",
    "QUADRICEPS" to "Ноги",
    "ABS" to "Пресс",
    "BICEPS" to "Руки",
    "CARDIO" to "Кардио",
    "FULL_BODY" to "Всё тело",
)

@Composable
fun MuscleGroupFilterChips(
    selected: String?,
    onSelected: (String?) -> Unit,
    modifier: Modifier = Modifier,
    contentPadding: PaddingValues = PaddingValues(horizontal = 20.dp),
) {
    Row(
        modifier = modifier
            .horizontalScroll(rememberScrollState())
            .padding(contentPadding),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        MuscleGroups.forEach { (value, label) ->
            val isSelected = selected == value
            Text(
                text = label,
                fontSize = 12.sp,
                fontWeight = FontWeight.SemiBold,
                color = if (isSelected) Color.White else DeepInk,
                modifier = Modifier
                    .clip(RoundedCornerShape(50))
                    .background(if (isSelected) AccentOrange else SoftCard)
                    .clickable { onSelected(value) }
                    .padding(horizontal = 14.dp, vertical = 8.dp),
            )
        }
    }
}

File: gymgenie-mobile/iosApp/iosApp/Components/MuscleGroupFilterChips.swift
import SwiftUI

struct MuscleGroupFilterChips: View {
    let selected: String?
    let onSelected: (String?) -> Void

    private let orange = Color(red: 0.941, green: 0.439, blue: 0.188)
    private let deepInk = Color(red: 0.161, green: 0.141, blue: 0.125)
    private let softCard = Color(red: 0.953, green: 0.949, blue: 0.937)

    private let groups: [(String?, String)] = [
        (nil, "Все"),
        ("CHEST", "Грудь"),
        ("BACK", "Спина"),
        ("SHOULDERS", "Плечи"),
        ("QUADRICEPS", "Ноги"),
        ("ABS", "Пресс"),
        ("BICEPS", "Руки"),
        ("CARDIO", "Кардио"),
        ("FULL_BODY", "Всё тело"),
    ]

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 8) {
                ForEach(Array(groups.enumerated()), id: \.offset) { _, item in
                    chip(value: item.0, label: item.1)
                }
            }
            .padding(.horizontal, 20)
        }
        .padding(.vertical, 4)
    }

    @ViewBuilder
    private func chip(value: String?, label: String) -> some View {
        let isSelected = value == selected
        Button(action: { onSelected(value) }) {
            Text(label)
                .font(.system(size: 12, weight: .semibold))
                .foregroundColor(isSelected ? .white : deepInk)
                .padding(.horizontal, 14)
                .padding(.vertical, 8)
                .background(Capsule().fill(isSelected ? orange : softCard))
        }
        .buttonStyle(.plain)
    }
}
